akka-http-backend/src/main/scala/sttp/client3/akkahttp/AkkaHttpBackend.scala

```diff
@@ -54,9 +54,13 @@ class AkkaHttpBackend private (
       .fromTry(ToAkka.request(r).flatMap(BodyToAkka(r, r.body, _)))
       .map(customizeRequest)
       .flatMap(request =>
-        http.singleRequest(request, connectionSettings(r)).map(response => customizeResponse(request, response))
+        http
+          .singleRequest(request, connectionSettings(r))
+          .flatMap(response =>
+            Future.apply(customizeResponse(request, response)).recoverWith(consumeResponseOnFailure(response))
```

Comment:
You can combine these two cases to recover only once, by flatMapping this future with the `responseFromAkka` funciton

Reply:
you mean that I can perform _recovery_ after the last `.flatMap` operation and as a result end up with:
```
      ...
      .flatMap(request =>
        http.singleRequest(request, connectionSettings(r)).map(response => customizeResponse(request, response))
      )
      .flatMap(response => responseFromAkka(r, response, None).recoverWith(consumeResponseOnFailure(response)))
```
right?

Reply:
Done